examples/regression/4-tutorials/plot_crf_tutorial.py

```diff
@@ -0,0 +1,274 @@
+            prefit=True
+        )
+    },
+}
+
+y_pred, intervals, coverage, cond_coverage = {}, {}, {}, {}
+num_bins = 10
+alpha = 0.1
+for strategy, params in STRATEGIES.items():
+    mapie = MapieRegressor(**params, random_state=random_state)
+    if mapie.conformity_score.prefit:
+        mapie.fit(X_calib_prefit, y_calib_prefit)
+    else:
+        mapie.fit(X_calib, y_calib)
+    y_pred[strategy], intervals[strategy] = mapie.predict(X_test, alpha=alpha)
+
+    coverage[strategy] = regression_coverage_score_v2(
+        y_test, intervals[strategy]
+    )
+    cond_coverage[strategy] = regression_ssc_score(
+        y_test, intervals[strategy], num_bins=num_bins
+    )
+
+
+def yerr(y_pred, intervals) -> ArrayLike:
+    """
+    Returns the error bars with the point prediction and its interval
+
+    Parameters
+    ----------
+    y_pred: ArrayLike
+        Point predictions.
+    intervals: ArrayLike
+        Predictions intervals.
+
+    Returns
+    -------
+    ArrayLike
+        Error bars.
+    """
+    return np.abs(np.concatenate(
+        [
+            np.expand_dims(y_pred, 0) - intervals[:, 0, 0].T,
+            intervals[:, 1, 0].T - np.expand_dims(y_pred, 0),
+        ],
+        axis=0,
+    ))
+
+
+def plot_predictions(y, y_pred, intervals, coverage, cond_coverage, ax=None):
+    """
+    Plots y_true against y_pred with the associated interval.
+
+    Parameters
+    ----------
+    y: ArrayLike
+       Observed targets
+    y_pred: ArrayLike
+       Predictions
+    intervals: ArrayLike
+       Prediction intervals
+    coverage: float
+       Global coverage
+    cond_coverage: float
+        Maximum violation coverage
+    ax: matplotlib axes
+       An ax can be provided to include this plot in a subplot
+    """
+    if ax is None:
+        fig, ax = plt.subplots(figsize=(6, 5))
+
+    ax.set_xlim([-0.5, 5.5])
+    ax.set_ylim([-0.5, 5.5])
+    ax.errorbar(
+        y,
+        y_pred,
+        yerr=yerr(y_pred, intervals),
+        color="#ff7f0e",
+        alpha=0.1,
+        linestyle="None",
+    )
+    ax.scatter(y, y_pred, s=2, color="#1f77b4")
+    ax.plot([0, max(max(y), max(y_pred))], [0, max(max(y), max(y_pred))], "-r")
+    ax.set_title(
+        f"{strategy} - coverage={coverage:.0%} " +
+        f"- max violation={cond_coverage:.0%}"
+    )
+    ax.set_xlabel("y true")
+    ax.set_ylabel("y pred")
+    ax.grid()
+
+
+fig, axs = plt.subplots(nrows=2, ncols=2, figsize=(12, 10))
+for ax, strategy in zip(axs.flat, STRATEGIES.keys()):
+    plot_predictions(
+        y_test,
+        y_pred[strategy],
+        intervals[strategy],
+        coverage[strategy][0],
+        cond_coverage[strategy][0],
+        ax=ax
+    )
+
+fig.suptitle(f"Predicted values and intervals of level {alpha}")
+plt.tight_layout()
+plt.show()
+
+##############################################################################
+# The results show that all the setups reach the global coverage guaranteed of
```

Comment:
I would be interested to hear more about the differences between each model, like now we just see, but it's not explained, could there be an explanation for the results observed?

Reply:
Do you want a description of the results of each method ? The purpose of this tutorial is more to show all the different possible ways to use CRF score and they are detailed under "2-Models"

Reply:
Indeed. I think it would be nice to have at least a little insight into the results. Like in two sentences mention some differences?